### DISCUSSION TITLE
Need to verify: Can Paste accept Custom Icons within the Components

### DISCUSSION THREAD

**Discussion by brannytempy** [twilio-labs/paste#2736] [Q&A] [answered]
Hi Paste!

I am working through the migration strategy for Segment. One large area of concern I have is about icons. Currently Segment has hundreds of icons (501 icons) that are stylistically different than Paste's. Many of your icons are outlined, where Segment prefers filled icons.

Because of this, I am considering recommending that Segment defers onboarding to Paste icons, until I can work with you all to get  1:1 Streamline icons ready and contributed and have us all migrate at once towards the end. My thinking is we will most definitely introduce further visual fragmentation by mixing 2 sets of icons.

I was pleased to see in some docs like [Icon Buttons](https://paste.twilio.design/components/button#icon-buttons), the Icon component itself is nested inside the component versus in a prop. Can you confirm that Segment can use their own icons from Evergreen in Paste components until a time as we're ready to switch? Are there any concerns, esp about tokens functioning with them?

Thx!!

**Comment by TheSisb** [comment, marked as answer]
Hi @brannytempy,

There are a few instances where icons are integral to component functionality:
- [Alert](https://paste.twilio.design/components/alert) (status icons - filled)
- [Callout](https://paste.twilio.design/components/callout) (status icons - filled)
- [Help Text](https://paste.twilio.design/components/help-text) (status icons - filled)
- [Checkbox](https://paste.twilio.design/components/checkbox) (the check)
- [Codeblock](https://paste.twilio.design/components/code-block/) (copy & link icon)
- [Combobox](https://paste.twilio.design/components/multiselect-combobox) (down caret and check)
- [Disclosure](https://paste.twilio.design/components/disclosure) (arrow)
- [Form Pill Group](https://paste.twilio.design/components/form-pill-group) (the X to dismiss)
- [Menu](https://paste.twilio.design/components/menu) (down caret)
- [Pagination](https://paste.twilio.design/components/pagination) (the arrows)
- [Popover](https://paste.twilio.design/components/popover) (the X to dismiss)
- [Modal](https://paste.twilio.design/components/modal) (the X to dismiss)
- [Toast](https://paste.twilio.design/components/toast) (the X to dismiss)


Going through this list, I think the migration will be fairly unobtrusive. The main ones that would be concerning are the status icons in the likes of Alert, but those are filled styled already.  The X to close icons[ look close enough already](https://evergreen.segment.com/foundations/icons#basic_icon_usage).  The rest seem fairly standard as well.

Hope this helps!
